src/k8s/util/enhancedK8sMethods.js

```diff
@@ -0,0 +1,121 @@
+import { get } from 'lodash';
+
+import { K8sCreateError, K8sKillError, K8sMultipleErrors, K8sPatchError } from './errors';
+import { findModel } from '../../models';
+import { getFullResourceId } from '../../utils';
+
+export const HISTORY_TYPE_CREATE = 'create';
+export const HISTORY_TYPE_PATCH = 'patch';
+export const HISTORY_TYPE_DELETE = 'delete';
+export const HISTORY_TYPE_NOT_FOUND = 'notFound';
+
+export class HistoryItem {
+  constructor(type, object) {
+    this.type = type;
+    this.object = object;
+  }
+}
+
+export class EnhancedK8sMethods {
+  constructor({ k8sCreate, k8sPatch, k8sKill }) {
+    this._history = [];
+    this._k8sCreate = k8sCreate;
+    this._k8sPatch = k8sPatch;
+    this._k8sKill = k8sKill;
+  }
+
+  _appendHistory = (historyItem, enhancedOpts) => {
+    if (!enhancedOpts || !enhancedOpts.disableHistory) {
```

Comment:
The use of `enhancedOpts` is confusing:
-  effectively, it's boolean. What about `isSomething` naming?
- based on its name, I would expect that it "turns on"  the feature. In fact, it disables it.

Reply:
it is called `enhancedOpt` because there is already one set of opts for the request. It is an object so it can be easier to extend in the future.

`disableHistory` is a command to the request. It just tells the request to not store history of it. `IsHistoryDisabled` doesn't sound right to me. 

\+ history is on by default, because that is the meaning of this class

Better name doesn't come to my mind...

Reply:
ok
Honestly, I really hope we will not need to farther decorate these low-level function. Sounds reasonable now, but might become nightmare when overused.